controllers/terminal/controllers/terminal_controller.go

```diff
@@ -207,6 +241,33 @@ func (r *TerminalReconciler) syncDeployment(ctx context.Context, terminal *termi
 	return nil
 }
 
+func (r *TerminalReconciler) fillDefaultValue(ctx context.Context, terminal *terminalv1.Terminal) error {
+	if terminal.Spec.APIServer == "" {
+		terminal.Spec.APIServer = DefaultAPIServer
+		return r.Update(ctx, terminal)
+	}
+	return nil
+}
+
+// isExpired return true if the terminal has expired
+func isExpired(terminal *terminalv1.Terminal) bool {
+	anno := terminal.ObjectMeta.Annotations
+	timeStr, exist := anno[KeepaliveAnnotation]
+	if !exist {
```

Comment:
If not exist we need set this annotations

Reply:
Or just return true to delete this terminal, must set this annotions when create the terminal.